src/OpenIddict.Core/Infrastructure/OpenIddictProvider.Authentication.cs

```diff
@@ -13,16 +13,113 @@
+
+                return;
+            }
+
+            // Reject requests using the unsupported request_uri parameter.
+            if (!string.IsNullOrEmpty(context.Request.RequestUri)) {
+                services.Logger.LogError("The authorization request was rejected because it contained " +
+                                         "an unsupported parameter: {Parameter}.", "request_uri");
+
+                context.Reject(
+                    error: OpenIdConnectConstants.Errors.RequestUriNotSupported,
+                    description: "The request_uri parameter is not supported.");
+
+                return;
+            }
+
+            // If a request_id parameter can be found in the authorization request,
+            // restore the complete authorization request stored in the user session.
+            if (!string.IsNullOrEmpty(context.Request.GetRequestId())) {
+                // Ensure session support has been enabled for this request.
+                if (context.HttpContext.Features.Get<ISessionFeature>() == null) {
+                    services.Logger.LogError("The authorization request was rejected because the session middleware " +
+                                             "was not correctly registered. Session support must be enabled to allow " +
+                                             "processing authorization requests specifying a request_id parameter.");
+
+                    context.Reject(
+                        error: OpenIdConnectConstants.Errors.InvalidRequest,
+                        description: "The request_id parameter is not allowed by this authorization server.");
+
+                    return;
+                }
+
+                // Load the session from the session store.
+                await context.HttpContext.Session.LoadAsync();
+
+                var payload = context.HttpContext.Session.Get("openiddict-request:" + context.Request.GetRequestId());
+                if (payload == null) {
+                    services.Logger.LogError("The authorization request was rejected because an unknown " +
+                                             "or invalid request_id parameter was specified.");
+
+                    context.Reject(
+                        error: OpenIdConnectConstants.Errors.InvalidRequest,
+                        description: "Invalid request: timeout expired.");
+
+                    return;
+                }
+
+                // Restore the authorization request parameters from the serialized payload.
+                context.Request.Import(payload);
+            }
+        }
+
         public override async Task ValidateAuthorizationRequest([NotNull] ValidateAuthorizationRequestContext context) {
             var services = context.HttpContext.RequestServices.GetRequiredService<OpenIddictServices<TUser, TApplication, TAuthorization, TScope, TToken>>();
 
+            // Note: the OpenID Connect server middleware supports authorization code, implicit, hybrid,
```

Comment:
FYI, ASOS no longer requests that use non-standard `response_type`/`response_mode` to allow you to implement custom flows (similarly to what's allowed for `grant_type`). Since we don't want to support non-standard flows in OpenIddict, I've moved these checks from ASOS to ensure we reject them properly.